### PR TITLE
fix(react-card): CardHeader grid layout ignoring line-height of content

### LINKED ISSUE #28890: [Bug]: Card layout, icon and text misaligned
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
Fluent v9
```

### Are you reporting Accessibility issue?

no

### Reproduction

https://aka.ms/fluentui-storybook

## Actual Behavior
image
image

Secondary text lineheight should be 16px (not sure why it's coming out to 20px here, causing this text unit to not be vertically centered w/ the File icon)
image

Overflow icon should be the Regular weight icon at rest (not filled). 
image

## Expected Behavior

Expected visual
image

File icon should be 32x32, I overwrote it in inspect mode here:
image

Secondary text lineheight should be 16px

### Logs

_No response_

### Requested priority

High

### Products/sites affected

Teams

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by marcosmoura**
CardHeader was ignoring the content line-height by setting `grid-auto-rows`. By removing it, it automatically uses the content size to determine the height of each row.

- Fixes #28890

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28968/perf-test-react-components/Avatar-mount_base.html">607</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28968/perf-test-react-components/Avatar-mount.html">602</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28968/perf-test-react-components/Button-mount_base.html">320</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28968/perf-test-react-components/Button-mount.html">307</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28968/perf-test-react-components/Field-mount_base.html">1108</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28968/perf-test-react-components/Field-mount.html">1115</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28968/perf-test-react-components/FluentProvider-mount_base.html">684</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28968/perf-test-react-components/FluentProvider-mount.html">680</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28968/perf-test-react-components/FluentProviderWithTheme-mount_base.html">81</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28968/perf-test-react-components/FluentProviderWithTheme-mount.html">92</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28968/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">68</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28968/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">68</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28968/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">70</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28968/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">71</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28968/perf-test-react-components/InfoButton-mount_base.html">13</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28968/perf-test-react-components/InfoButton-mount.html">15</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28968/perf-test-react-components/MakeStyles-mount_base.html">880</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28968/perf-test-react-components/MakeStyles-mount.html">863</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28968/perf-test-react-components/Persona-mount_base.html">1746</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28968/perf-test-react-components/Persona-mount.html">1659</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28968/perf-test-react-components/SpinButton-mount_base.html">1394</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28968/perf-test-react-components/SpinButton-mount.html">1335</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                                                                             |      Baseline (minified/GZIP) |                            PR |                                                                                                                                                                                                  Change |
| :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: | ----------------------------: | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.All.fixture.js'>Card - All</abbr>                                                                                |  `88.335 kB`<br />`25.271 kB` |  `88.334 kB`<br />`25.252 kB` | `-1 B` <br />`-19 B`  |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardHeader.fixture.js'>CardHeader</abbr>                                                                              |    `11.59 kB`<br />`4.716 kB` |   `11.589 kB`<br />`4.707 kB` |  `-1 B` <br />`-9 B`  |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `203.781 kB`<br />`57.916 kB` | `203.786 kB`<br />`57.921 kB` |    `5 B` <br />`5 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                |         Size (minified/GZIP) |
| ---------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.fixture.js'>Card</abbr>                                                                             | `83.177 kB`<br />`23.742 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardFooter.fixture.js'>CardFooter</abbr>                                                                 |   `9.338 kB`<br />`3.932 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardPreview.fixture.js'>CardPreview</abbr>                                                               |  `10.297 kB`<br />`4.317 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr> | `66.614 kB`<br />`18.512 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>               | `37.787 kB`<br />`12.387 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                            |    `6.48 kB`<br />`2.203 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/afcfb2ebe322e2a8e204ac192e68f48f1164107f'>afcfb2ebe322e2a8e204ac192e68f48f1164107f</a></sub>
<div id="bundleSizeComment9423"/>

File: change/@fluentui-react-card-97d7abde-06d9-46cc-a71a-147630a0431e.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-card-97d7abde-06d9-46cc-a71a-147630a0431e.json:1]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=28968&commitId=eb3032532b877f21cbac0d091b388ed6a45bbbae&env=prod&clientType=FLUENTUI)


  <details><summary><b>Card Converged</b> 8 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Card Converged.card templates.normal.chromium.png|2774| Changed|
|Card Converged.appearance.normal.chromium.png|35403| Changed|
|Card Converged.CardHeader.normal.chromium.png|11378| Changed|
|Card Converged.appearance - High Contrast.normal.chromium.png|40980| Changed|
|Card Converged.appearance - Dark Mode.normal.chromium.png|24215| Changed|
|Card Converged.appearance - RTL.normal.chromium.png|35411| Changed|
|Card Converged.size.normal.chromium.png|23648| Changed|
|Card Converged.orientation.chromium.png|6216| Changed|

</details>


  <details><summary><b>Card Converged - Interactive</b> 36 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Card Converged - Interactive.appearance interactive - Filled - High Contrast.click.chromium.png|3387| Changed|
|Card Converged - Interactive.appearance interactive - Filled - Dark Mode.normal.chromium.png|3292| Changed|
|Card Converged - Interactive.appearance interactive - Filled - Dark Mode.click.chromium.png|3402| Changed|
|Card Converged - Interactive.appearance interactive - Filled Alternative.normal.chromium.png|4611| Changed|
|Card Converged - Interactive.appearance interactive - Filled - High Contrast.normal.chromium.png|5269| Changed|
|Card Converged - Interactive.appearance interactive - Filled Alternative - High Contrast.click.chromium.png|3387| Changed|
|Card Converged - Interactive.appearance interactive - Filled.click.chromium.png|4160| Changed|
|Card Converged - Interactive.appearance interactive - Filled - Dark Mode.hover.chromium.png|3121| Changed|
|Card Converged - Interactive.appearance interactive - Filled Alternative - High Contrast.normal.chromium.png|5269| Changed|
|Card Converged - Interactive.appearance interactive - Filled Alternative.click.chromium.png|4066| Changed|
|Card Converged - Interactive.appearance interactive - Filled - High Contrast.hover.chromium.png|3387| Changed|
|Card Converged - Interactive.appearance interactive - Filled Alternative - Dark Mode.normal.chromium.png|3402| Changed|
|Card Converged - Interactive.appearance interactive - Filled Alternative - Dark Mode.hover.chromium.png|3187| Changed|
|Card Converged - Interactive.appearance interactive - Filled.normal.chromium.png|4712| Changed|
|Card Converged - Interactive.appearance interactive - Outline - High Contrast.click.chromium.png|3198| Changed|
|Card Converged - Interactive.appearance interactive - Outline - High Contrast.hover.chromium.png|3198| Changed|
|Card Converged - Interactive.appearance interactive - Outline - Dark Mode.click.chromium.png|3449| Changed|
|Card Converged - Interactive.appearance interactive - Outline - High Contrast.normal.chromium.png|5269| Changed|
|Card Converged - Interactive.appearance interactive - Subtle - Dark Mode.normal.chromium.png|3293| Changed|
|Card Converged - Interactive.appearance interactive - Outline - Dark Mode.hover.chromium.png|3472| Changed|
|Card Converged - Interactive.appearance interactive - Outline - Dark Mode.normal.chromium.png|3437| Changed|
|Card Converged - Interactive.appearance interactive - Filled Alternative.hover.chromium.png|4444| Changed|
|Card Converged - Interactive.appearance interactive - Subtle.hover.chromium.png|4513| Changed|
|Card Converged - Interactive.appearance interactive - Subtle - High Contrast.normal.chromium.png|5269| Changed|
|Card Converged - Interactive.appearance interactive - Subtle - Dark Mode.hover.chromium.png|3164| Changed|
|Card Converged - Interactive.appearance interactive - Outline.normal.chromium.png|4712| Changed|
|Card Converged - Interactive.appearance interactive - Outline.click.chromium.png|4713| Changed|
|Card Converged - Interactive.appearance interactive - Subtle.click.chromium.png|4161| Changed|
|Card Converged - Interactive.appearance interactive - Filled Alternative - High Contrast.hover.chromium.png|3387| Changed|
|Card Converged - Interactive.appearance interactive - Outline.hover.chromium.png|4713| Changed|
|Card Converged - Interactive.appearance interactive - Subtle.normal.chromium.png|4712| Changed|
|Card Converged - Interactive.appearance interactive - Subtle - Dark Mode.click.chromium.png|3240| Changed|
|Card Converged - Interactive.appearance interactive - Subtle - High Contrast.hover.chromium.png|3387| Changed|
|Card Converged - Interactive.appearance interactive - Filled Alternative - Dark Mode.click.chromium.png|3566| Changed|
|Card Converged - Interactive.appearance interactive - Filled.hover.chromium.png|4513| Changed|
|Card Converged - Interactive.appearance interactive - Subtle - High Contrast.click.chromium.png|3387| Changed|

</details>


  <details><summary><b>Card Converged - Selectable</b> 68 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Card Converged - Selectable.appearance focusable + selectable.normal.chromium.png|4712| Changed|
|Card Converged - Selectable.appearance focusable + selectable.selected.chromium.png|4356| Changed|
|Card Converged - Selectable.appearance selectable - Filled - Dark Mode.click.chromium.png|3402| Changed|
|Card Converged - Selectable.appearance selectable - Filled - RTL.click.chromium.png|4156| Changed|
|Card Converged - Selectable.appearance selectable - Filled - Dark Mode.selected.chromium.png|3297| Changed|
|Card Converged - Selectable.appearance selectable - Filled - High Contrast.selected.chromium.png|3198| Changed|
|Card Converged - Selectable.appearance selectable - Filled - High Contrast.normal.chromium.png|5269| Changed|
|Card Converged - Selectable.appearance selectable - Filled - RTL.selected.chromium.png|4352| Changed|
|Card Converged - Selectable.appearance selectable - Filled - Dark Mode.normal.chromium.png|3292| Changed|
|Card Converged - Selectable.appearance selectable - Filled - RTL.hover.chromium.png|4507| Changed|
|Card Converged - Selectable.appearance selectable - Filled Alternative - Dark Mode.click.chromium.png|3566| Changed|
|Card Converged - Selectable.appearance selectable - Filled - Dark Mode.hover.chromium.png|3121| Changed|
|Card Converged - Selectable.appearance selectable - Filled Alternative - RTL.click.chromium.png|4061| Changed|
|Card Converged - Selectable.appearance selectable - Filled Alternative - High Contrast.normal.chromium.png|5269| Changed|
|Card Converged - Selectable.appearance focusable + selectable.hover.chromium.png|4513| Changed|
|Card Converged - Selectable.appearance selectable - Filled Alternative - Dark Mode.hover.chromium.png|3187| Changed|
|Card Converged - Selectable.appearance selectable - Filled Alternative - RTL.hover.chromium.png|4439| Changed|
|Card Converged - Selectable.appearance focusable + selectable.click.chromium.png|4160| Changed|
|Card Converged - Selectable.appearance selectable - Filled - RTL.normal.chromium.png|4707| Changed|
|Card Converged - Selectable.appearance selectable - Filled Alternative - High Contrast.hover.chromium.png|3387| Changed|
|Card Converged - Selectable.appearance selectable - Filled - High Contrast.hover.chromium.png|3387| Changed|
|Card Converged - Selectable.appearance selectable - Filled Alternative - High Contrast.click.chromium.png|3387| Changed|
|Card Converged - Selectable.appearance selectable - Filled Alternative - RTL.selected.chromium.png|4263| Changed|
|Card Converged - Selectable.appearance selectable - Filled Alternative.normal.chromium.png|4611| Changed|
|Card Converged - Selectable.appearance selectable - Filled.hover.chromium.png|4513| Changed|
|Card Converged - Selectable.appearance selectable - Filled.selected.chromium.png|4356| Changed|
|Card Converged - Selectable.appearance selectable - Filled Alternative - RTL.normal.chromium.png|4606| Changed|
|Card Converged - Selectable.appearance selectable - Filled.click.chromium.png|4160| Changed|
|Card Converged - Selectable.appearance selectable - Filled Alternative.click.chromium.png|4066| Changed|
|Card Converged - Selectable.appearance selectable - Filled Alternative - High Contrast.selected.chromium.png|3198| Changed|
|Card Converged - Selectable.appearance selectable - Filled Alternative - Dark Mode.selected.chromium.png|3396| Changed|
|Card Converged - Selectable.appearance selectable - Outline - Dark Mode.click.chromium.png|3449| Changed|
|Card Converged - Selectable.appearance selectable - Outline - Dark Mode.selected.chromium.png|3472| Changed|
|Card Converged - Selectable.appearance selectable - Outline - Dark Mode.normal.chromium.png|3437| Changed|
|Card Converged - Selectable.appearance selectable - Filled Alternative.selected.chromium.png|4268| Changed|
|Card Converged - Selectable.appearance selectable - Filled - High Contrast.click.chromium.png|3387| Changed|
|Card Converged - Selectable.appearance selectable - Filled Alternative.hover.chromium.png|4444| Changed|
|Card Converged - Selectable.appearance selectable - Outline - High Contrast.click.chromium.png|3198| Changed|
|Card Converged - Selectable.appearance selectable - Filled.normal.chromium.png|4712| Changed|
|Card Converged - Selectable.appearance selectable - Outline - High Contrast.selected.chromium.png|3198| Changed|
|Card Converged - Selectable.appearance selectable - Outline - RTL.click.chromium.png|4708| Changed|
|Card Converged - Selectable.appearance selectable - Outline.click.chromium.png|4713| Changed|
|Card Converged - Selectable.appearance selectable - Subtle - Dark Mode.click.chromium.png|3240| Changed|
|Card Converged - Selectable.appearance selectable - Subtle - Dark Mode.normal.chromium.png|3293| Changed|
|Card Converged - Selectable.appearance selectable - Filled Alternative - Dark Mode.normal.chromium.png|3402| Changed|
|Card Converged - Selectable.appearance selectable - Outline - High Contrast.hover.chromium.png|3198| Changed|
|Card Converged - Selectable.appearance selectable - Outline - High Contrast.normal.chromium.png|5269| Changed|
|Card Converged - Selectable.appearance selectable - Outline.selected.chromium.png|4713| Changed|
|Card Converged - Selectable.appearance selectable - Subtle - High Contrast.selected.chromium.png|3198| Changed|
|Card Converged - Selectable.appearance selectable - Outline.normal.chromium.png|4712| Changed|
|Card Converged - Selectable.appearance selectable - Outline - RTL.selected.chromium.png|4707| Changed|
|Card Converged - Selectable.appearance selectable - Outline.hover.chromium.png|4713| Changed|
|Card Converged - Selectable.appearance selectable - Outline - RTL.hover.chromium.png|4707| Changed|
|Card Converged - Selectable.appearance selectable - Subtle - High Contrast.normal.chromium.png|5269| Changed|
|Card Converged - Selectable.appearance selectable - Subtle - RTL.hover.chromium.png|4508| Changed|
|Card Converged - Selectable.appearance selectable - Subtle.normal.chromium.png|4712| Changed|
|Card Converged - Selectable.appearance selectable - Subtle - RTL.normal.chromium.png|4707| Changed|
|Card Converged - Selectable.appearance selectable - Subtle.click.chromium.png|4161| Changed|
|Card Converged - Selectable.appearance selectable - Subtle.hover.chromium.png|4513| Changed|
|Card Converged - Selectable.appearance selectable - Subtle - High Contrast.click.chromium.png|3387| Changed|
|Card Converged - Selectable.appearance selectable - Outline - Dark Mode.hover.chromium.png|3472| Changed|
|Card Converged - Selectable.appearance selectable - Subtle - RTL.click.chromium.png|4157| Changed|
|Card Converged - Selectable.appearance selectable - Subtle.selected.chromium.png|4357| Changed|
|Card Converged - Selectable.appearance selectable - Subtle - Dark Mode.hover.chromium.png|3164| Changed|
|Card Converged - Selectable.appearance selectable - Outline - RTL.normal.chromium.png|4707| Changed|
|Card Converged - Selectable.appearance selectable - Subtle - High Contrast.hover.chromium.png|3387| Changed|
|Card Converged - Selectable.appearance selectable - Subtle - Dark Mode.selected.chromium.png|3331| Changed|
|Card Converged - Selectable.appearance selectable - Subtle - RTL.selected.chromium.png|4353| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Review by tudorpopams — APPROVED** [review]

File: change/@fluentui-react-card-97d7abde-06d9-46cc-a71a-147630a0431e.json (hunk 1 above)
Comment (marcosmoura) [reply change/@fluentui-react-card-97d7abde-06d9-46cc-a71a-147630a0431e.json:1]: Resolved as this is intended to happen. The card header now has less space between the elements which result in the whole rest of the content shifting to the top.

**Pull request merged**